### PR TITLE
[Feature] Week 3 Features

### PR DESCRIPTION
**Pull request by daekoon**
## Changes
- Added support for user authentication
- Comments no longer show if user is not logged in
- Replaced the "Username" field with "Email"
- Added feature to detect the current user's email and store it with the comment
- Display current logged in user's email

### CODE REVIEW

File: portfolio/src/main/webapp/commentScript.js [hunk 1]
```diff
@@ -15,25 +15,42 @@
 document.addEventListener('DOMContentLoaded', displayComments(), false);
 
 function displayComments() {
-  fetch('/comments').then(response => response.json()).then(comments => {
-    const commentsContainer = document.getElementById('comments-container');
-    var html = '<table id="commentsTable" width="80%">';
+  fetch('/comments').then(response => response.json()).then(response => {
+    const commentsContainer = document.getElementById('content');
+    var html = '';
+    if (!response.loggedIn) {
```
Comment (henryken) [inline portfolio/src/main/webapp/commentScript.js:21]: Is it intended that user can't see any comment when not logged in?

Comment (daekoon) [reply portfolio/src/main/webapp/commentScript.js:21]: Yes, it is intentional.

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 2]
```diff
@@ -36,38 +39,61 @@
...
 
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  UserService userService = UserServiceFactory.getUserService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  
+    response.setContentType("application/json");
     
+    Gson gson = new Gson();
```
[the remark this answers is not in the document]
Comment (daekoon) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:49]: The Gson is used below as well (line 81)

[the remark this answers is not in the document]
Comment (henryken) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:49]: Okay I missed that. Noted.

File: portfolio/src/main/java/com/google/sps/data/CommentsResponse.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.google.sps.data;
+
+import java.util.List;
+
+/**
+ * Class that is used as a JSON Response to the /comments GET endpoint
+ */
+
+public class CommentsResponse {
```
[the remark this answers is not in the document]
Comment (daekoon) [reply portfolio/src/main/java/com/google/sps/data/CommentsResponse.java:9]: Removed login related information from `CommentsResponse`

**Comment by daekoon** [comment]
Took me a while to decouple the login and comment logic, but its finally done

File: portfolio/src/main/java/com/google/sps/data/CommentsResponse.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    private List<Comment> comments;
+    private String currentUserEmail;
```
Comment (henryken) [inline portfolio/src/main/java/com/google/sps/data/CommentsResponse.java:11]: I think this is better to include as part of the LoginResponse, as it is part of the user authentication information.

**Comment by henryken** [comment]
Thanks for making the changes. I have one comment. Let me know whether it's doable.

File: portfolio/src/main/java/com/google/sps/data/CommentsResponse.java (hunk 4 above)
Comment (daekoon) [reply portfolio/src/main/java/com/google/sps/data/CommentsResponse.java]: Moved `currentUserEmail` to `LoginResponse`

**Comment by daekoon** [comment]
Managed to further decouple the logic :)

**Pull request merged**